internal/background/availability_queue.go

```diff
@@ -31,12 +40,36 @@ func sendAvailabilityRequestMessages(ctx context.Context, batchSize int, tickDur
 	logger := ctxval.Logger(ctx)
 	ticker := time.NewTicker(tickDuration)
 	messageBuffer := make([]*kafka.GenericMessage, 0, batchSize)
+	sc, err := clients.GetSourcesClient(ctx)
+	if err != nil {
+		logger.Warn().Msg("Could not get sources client")
+		return
+	}
 
 	for {
 		select {
-		case msg := <-kafkaAvailabilityRequest:
-			messageBuffer = append(messageBuffer, msg)
+		case info := <-kafkaAvailabilityRequest:
+			ctx, err = ctxval.WithIdentityFrom64(ctx, info.identity)
```

Comment:
We are redefining the context every time, I don't think it is what we want, I'd rather have a new variable for this context here, otherwise the identity can leak into other cases in this function and I don't think it is waht we want.